Speed up log prefix matching

janxious opened the pull request:
__ISSUE:__
_Performance_ - Filtering by prefix performance issues - Fixes #2

__DESCRIPTION:__
RT (and possibly other modpacks) are shipping with [large](https://github.com/BattletechModders/RogueTech/blob/72fdc89dbbddffd96b8d46099cd7c9a538bc073f/FYLS/mod.json#L18-L140) (121 currently)
log prefix filtering settings. The original method for filtering was not
intended for that size of prefixes (it worked fine with 2-10 certainly).
Regex is a better easy solution to filtering and may be good enough to
never need replacement or enhancement.

*Note: I haven't compiled this so there are probably some syntax errors.*

__STEPS TO TEST:__
Install new dll. Measure performance. @m22spencer seems likely to have
the ability to do this easily, but it should be noticeably less laggy.

__REFERENCES:__
0. [Discord Conversation about performance issue](https://discordapp.com/channels/565106671693856778/565134819894231040/740849129168830477)

__CHANGELIST:__
modified:   Core.cs

janxious committed:
refactor:2 speed up log prefix matching

__ISSUE:__
_Performance_ - Filtering by prefix performance issues - Fixes #2

__DESCRIPTION:__
RT (and possibly other modpacks) are shipping with large(121 currently)
log prefix filtering settings. The original method for filtering was not
intended for that size of prefixes (it worked fine with 2-10 certainly).
Regex is a better easy solution to filtering and may be good enough to
never need replacement or enhancement.

*Note: I haven't compiled this so there are probably some syntax errors.*

___STEPS TO TEST:__
Install new dll. Measure performance. @m22spencer seems likely to have
the ability to do this easily, but it should be noticeably less laggy.

__REFERENCES:__
0. [Discord Conversation about performance issue](https://discordapp.com/channels/565106671693856778/565134819894231040/740849129168830477)

__CHANGELIST:__
modified:   Core.cs

CptMoore commented:
If we want to get overboard with this, we need a trie regex data structure ... https://stackoverflow.com/a/56178401 😃

janxious committed:
update vsnet bits to work with current BTG, bump version to 1.4.0

__DESCRIPTION:__
vsnet gonna vsnet

__CHANGELIST:__
	modified:   BT-FYLS.sln
	modified:   FYLS.csproj
	modified:   mod.json
